src/Contract/Time.purs

```diff
@@ -98,10 +104,70 @@ import Ctl.Internal.Types.Interval
 import Ctl.Internal.Types.SystemStart (SystemStart)
 import Ctl.Internal.Types.SystemStart (SystemStart(SystemStart)) as ExportSystemStart
 import Data.BigInt as BigInt
+import Data.Foldable (find)
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Newtype (unwrap)
 import Data.UInt as UInt
+import Effect.Aff (delay)
 import Effect.Aff.Class (liftAff)
 import Effect.Exception (error)
 
+-- | Get a summary of the current era.
+currentEra :: Contract ExportEraSummaries.EraSummary
+currentEra = do
+  eraSummaries <- getEraSummaries
+  currentSlot <- getCurrentSlot
+  logInfo' $ show eraSummaries
+  logInfo' $ show currentSlot
+  -- Assumes that eras are sorted: this may not be stable in the future.
+  liftContractM "Could not find era summary"
+    $ find (go currentSlot)
+    $ unwrap eraSummaries
+  where
+  go :: Slot -> ExportEraSummaries.EraSummary -> Boolean
+  go currentSlot era =
+    let
+      eraStartSlot = era # unwrap # _.start # unwrap # _.slot
+      startNotAfterUs = eraStartSlot <= currentSlot
+    in
+      case era # unwrap # _.end of
+        Nothing -> startNotAfterUs
+        Just eraEnd -> startNotAfterUs &&
+          ( (eraEnd # unwrap # _.slot) >
+              currentSlot
+          )
+
+  getCurrentSlot :: Contract Slot
+  getCurrentSlot = do
+    { delay: delayMs } <- asks $ _.timeParams >>> _.awaitTxConfirmed
+    getChainTip >>= case _ of
+      Chain.TipAtGenesis -> do
+        liftAff $ delay delayMs
+        getCurrentSlot
+      Chain.Tip (Chain.ChainTip { slot }) -> pure slot
+
+-- | Given a desired range, tighten it to fit onchain.
+mkTimeRangeWithinSummary
```

Comment:
Please come up with a better name. Let the fallback option be `normalizeTimeInterval`